README.md

```diff
@@ -75,47 +75,143 @@ Configuration of these values is here (https://github.com/hmrc/upscan-infrastruc
+
+The service replies with a pre-filled template for the upload of the file.
+The JSON response also contains a globally unique file reference of the upload. This reference can be used by the Upscan service team to view the progress and result of the journey through the different Upscan components. The consuming service can use this reference to correlate the upload request with a successfully uploaded file.
+
+
+### POST v2/upscan/initiate
+
+The S3 rest api is able to redirect on successful uploads. It does not support redirects on errors however. Version 2 of `upscan/initiate` returns the url [upscan-upload-proxy](https://github.com/hmrc/upscan-upload-proxy) that sits in front of S3 to gracefully handle S3 errors.  
+
+Example `v2/upscan/initiate` request:
 
 ```json
 {
     "callbackUrl": "https://myservice.com/callback",
+    "successRedirect": "https://myservice.com/nextPage",
+    "errorRedirect": "https://myservice.com/errorPage",
     "minimumFileSize" : 0,
     "maximumFileSize" : 1024
 }
 ```
 
-Meaning of parameters:
+#### HTTP Headers:
+
+| Header name|Description|Required|
+|--------------|-----------|--------|
+| User-Agent | Identifier of the service that calls upscan | yes |
+| X-Session-ID | Identifier of the user's session | no  |
+| X-Request-ID | Identifier of the user's request | no |
+
+#### Body parameters:
 
 | Parameter name|Description|Required|
 |--------------|-----------|--------|
 |callbackUrl   |Url that will be called to report the outcome of file checking and upload, including retrieval details if successful. Notification format is detailed further down in this file. Must be https.| yes|
+|successRedirect|Url to redirect to after file has been successfully uploaded.|yes|
+|errorRedirect|Url to redirect to if error encountered during upload.|yes|
 |minimumFileSize|Minimum file size (in Bytes). Default is 0.|no|
 |maximumFileSize|Maximum file size (in Bytes). Cannot be greater than 100MB. Default is 100MB.|no|
-|successRedirect|Url to redirect to after file has been successfully uploaded.|no|
-The request has to include the following HTTP headers:
+
+Example response
+
+```json
+{
+    "reference": "11370e18-6e24-453e-b45a-76d3e32ea33d",
+    "uploadRequest": {
+        "href": "https://xxxx/upscan-upload-proxy/bucketName",
+        "fields": {
+            "Content-Type": "application/xml",
+            "acl": "private",
+            "key": "xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+            "policy": "xxxxxxxx==",
+            "x-amz-algorithm": "AWS4-HMAC-SHA256",
+            "x-amz-credential": "ASIAxxxxxxxxx/20180202/eu-west-2/s3/aws4_request",
+            "x-amz-date": "yyyyMMddThhmmssZ",
+            "x-amz-meta-callback-url": "https://myservice.com/callback",
+            "x-amz-signature": "xxxx",
+            "success_action_redirect": "https://myservice.com/nextPage",
+            "error_action_redirect": "https://myservice.com/errorPage"
```

Comment:
I would very explicitly mention that catalogue of these fields  might change.
Lot's of people have problem wtih that.